### ISSUE TITLE
dummy linux-new is always tried to download when CT_LINUX_CUSTOM_LOCATION is set

### ISSUE THREAD

**Issue by bagasme** [crosstool-ng/crosstool-ng#1511]
crosstool-NG version: commit f284f4149518de6e8c403a9392be8e817bfab2e8
host OS: Ubuntu 20.04

I was playing with `Custom source location` (experimental). So on config, I enabled `CT_LINUX_SRC_CUSTOM` and specified the custom source location path for linux tarball and then linux sources directory on `CT_LINUX_CUSTOM_LOCATION`. I also forbid any downloads (`CT_FORBID_DOWNLOAD`).

Examining build.log, I noticed strange behavior. I passed the CSL, so I expected that crosstool-NG would use linux sources from the specified path. In fact, dummy `linux-new` is instead to be downloaded. And because I had forbidden downloads, the build were terminated with error.

Below is the relevant log excerpt (on linux sources CSL):
```
[INFO ]  Retrieving needed toolchain components' tarballs
[DEBUG]    Package iterator: set basename=''
[DEBUG]    Package iterator: set pkg_name='linux'
[DEBUG]    Package iterator: set version='new'
[DEBUG]    Package iterator: set pkg_dir=''
[DEBUG]    Package iterator: set src_release='y'
[DEBUG]    Package iterator: set mirrors='-unknown-'
[DEBUG]    Package iterator: set archive_filename='@{pkg_name}-@{version}'
[DEBUG]    Package iterator: set archive_dirname='@{pkg_name}-@{version}'
[DEBUG]    Package iterator: set archive_formats='.tar.xz .tar.gz'
[DEBUG]    Package iterator: set signature_format='unpacked/.sign'
[DEBUG]    Package iterator: set src_devel=''
[DEBUG]    Package iterator: set devel_vcs=''
[DEBUG]    Package iterator: set devel_url=''
[DEBUG]    Package iterator: set devel_branch=''
[DEBUG]    Package iterator: set devel_revision=''
[DEBUG]    Package iterator: set devel_subdir=''
[DEBUG]    Package iterator: set devel_bootstrap=''
[DEBUG]    Package iterator: set src_custom='y'
[DEBUG]    Package iterator: set custom_location='/tmp/linux'
[DEBUG]    Package iterator: set patch_order='global'
[DEBUG]    Package iterator: run CT_DoFetch 
[DEBUG]    Testing '-n linux'
[DEBUG]    Not allowed to download from the Internet, aborting linux-new download
[ERROR]    linux: download failed
[ERROR]  
[ERROR]  >>
[ERROR]  >>  Build failed in step 'Retrieving needed toolchain components' tarballs'
[ERROR]  >>        called in step '(top-level)'
[ERROR]  >>
[ERROR]  >>  Error happened in: CT_Abort[scripts/functions@487]
[ERROR]  >>        called from: CT_DoFetch[scripts/functions@2131]
[ERROR]  >>        called from: CT_PackageRun[scripts/functions@2091]
[ERROR]  >>        called from: CT_Fetch[scripts/functions@2202]
[ERROR]  >>        called from: do_kernel_get[scripts/build/kernel/linux.sh@22]
[ERROR]  >>        called from: main[scripts/crosstool-NG.sh@646]
[ERROR]  >>
```
The problematic step was `test -n linux`. I expected that since I defined `CT_LINUX_CUSTOM_LOCATION`, the config's string should be non-zero.

I attached build log and config for each configuration (the tarball case and the directory case):

[build.linux-location.config.gz](https://github.com/crosstool-ng/crosstool-ng/files/6501593/build.linux-location.config.gz)
[build.tarball-location.config.gz](https://github.com/crosstool-ng/crosstool-ng/files/6501596/build.tarball-location.config.gz)
[linux-location.config.gz](https://github.com/crosstool-ng/crosstool-ng/files/6501598/linux-location.config.gz)
[tarball-location.config.gz](https://github.com/crosstool-ng/crosstool-ng/files/6501600/tarball-location.config.gz)

**Comment by bensuperpc** [comment]
I have the same problem :/

**Comment by cpackham** [comment]
We've ended up with both `CT_LINUX_SRC_CUSTOM=y` and `CT_LINUX_SRC_RELEASE=y` so the src_release path in `CT_DoFetch` is taken which naturally doesn't work.

**Comment by cpackham** [comment]
Should be fixed by #1527. I messed up the Fixes tag so the automatic link didn't happen
